### PR TITLE
Baca Manga Extension

### PR DESCRIPTION
**Pull request by rama982**
Indonesian

### CODE REVIEW

**Comment by as280093** [comment]
Here is the Icons use this 
[ic_launcher.zip](https://github.com/inorichi/tachiyomi-extensions/files/3716749/ic_launcher.1.zip)

**Comment by rama982** [comment]
> Here is the Icons use this
> [ic_launcher.zip](https://github.com/inorichi/tachiyomi-extensions/files/3716749/ic_launcher.1.zip)

thx updated

File: src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    private fun parseStatus(element: String): Int = when {
+        element.toLowerCase().contains("ongoing") -> SManga.ONGOING
+        element.toLowerCase().contains("completed") -> SManga.COMPLETED
+        else -> SManga.UNKNOWN
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val chapters = mutableListOf<SChapter>()
+        document.select(chapterListSelector()).map { chapters.add(chapterFromElement(it)) }
```
Comment (arkon) [inline src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt]: Similarly here:

```kotlin
val chapters = document.select(chapterListSelector()).map { chapterFromElement(it) }
```

File: src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+            .let {
+                val date = JSONObject(it).getJSONArray("@graph").
+                    getJSONObject(3).getString("dateModified")
+                chapters[0].date_upload = parseDate(date)
+            }
+        return chapters
+    }
+
+    private fun parseDate(date: String): Long {
+        return SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX").parse(date).time
```
Comment (arkon) [inline src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt]: Pass the locale in too.

File: src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        val sepName = infoElement.select("span:nth-child(4)").last()
+        val manga = SManga.create()
+        manga.author = sepName.ownText()
+        manga.artist = sepName.ownText()
+        val genres = mutableListOf<String>()
+        infoElement.select("span:nth-child(1) > a").forEach { element ->
+            val genre = element.text()
+            genres.add(genre)
+        }
+        manga.genre = genres.joinToString(", ")
```
[the remark this answers is not in the document]
Comment (rama982) [reply src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt]: Yes, i get the enlightenment

File: src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt (hunk 1 above)
Comment (rama982) [reply src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt]: i will change to that.

File: src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt (hunk 2 above)
Comment (rama982) [reply src/id/bacamanga/src/eu/kanade/tachiyomi/extension/id/bacamanga/BacaManga.kt]: Will pass

**Pull request merged**